solvers/elliptic/src/ellipticSettings.cpp

```diff
@@ -66,6 +66,29 @@ void ellipticAddSettings(settings_t& settings,
                       "Iterative Linear Solver to use for solve",
                       {"PCG", "FPCG", "NBPCG", "NBFPCG", "PGMRES"});
 
+  settings.newSetting(prefix+"LINEAR SOLVER STOPPING CRITERION",
+                      "ABS/REL-INITRESID",
+                      "Stopping criterion for the linear solver",
+                      {"ABS/REL-INITRESID", "ABS/REL-RHS-2NORM"});
+
+  settings.newSetting(prefix+"INITIAL GUESS STRATEGY",
+                      "NONE",
+                      "Strategy for selecting initial guess for linear solver",
+                      {"NONE", "ZERO", "CLASSIC", "QR", "EXTRAP"});
+
+  settings.newSetting(prefix+"INITIAL GUESS HISTORY SPACE DIMENSION",
+                      "-1",
+                      "Dimension of the initial guess space");
+
+  settings.newSetting(prefix+"INITIAL GUESS EXTRAP DEGREE",
+                      "-1",
+                      "Degree used for EXTRAP initial guess schemes.");
+
+  settings.newSetting(prefix+"INITIAL GUESS EXTRAP COEFFS METHOD",
+                      "MINNORM",
+                      "Method for selecting coefficients with EXTRAP initial guess schemes.",
+                      {"MINNORM", "CPQR"});
+
```

Comment:
Is using the initial guess machinery really part of the elliptic solver? I.e., should these settings be here, or should they be part of the larger solver (ins, fpe) that would construct them?

Reply:
Yeah, that's probably better, but there's a disadvantage to that:  if we were to add a new initial guess method, we would have to update `*Settings.cpp` in all the individual solvers to make it available to them.

Probably the right solution would be for the linear solver settings to be "owned" by `linearSolver_t` instead.

Reply:
In other instances where I find myself adding the same settings over and over again, I usually make an `*AddSettings` in the lib/namespace which I can reuse. Something like the parAlmond or elliptic AddSettings may work.